UI/TechTreeWnd.cpp

```diff
@@ -238,139 +255,155 @@ class TechTreeWnd::TechTreeControls : public CUIWnd {
     // calculated by SizeMove, and stored, so that start and end positions don't need to be recalculated each
     // time Render is called.
 
-    std::vector<CUIButton*>             m_category_buttons;
-    std::map<TechStatus, CUIButton*>    m_tech_status_buttons;
-    CUIButton*                          m_list_view_button;
-    CUIButton*                          m_tree_view_button;
+    boost::shared_ptr<GG::SubTexture>            GetManagedSubTexture(const std::string& fn);  ///< return a shared_ptr of a managed subtexture for tech controls
+    CUIToggleButton*                             m_view_type_button;
+    CUIToggleRepresenter*                        m_view_type_representer;
+    CUIToggleButton*                             m_all_cat_button;
+    CUIToggleRepresenter*                        m_all_cat_representer;
```

Comment:
No need to store a pointer to a CUIToggleRepresenter, just initialize it and pass it to the CUIToggleButton.